docs: add docs for serialization of DocList (#1350)
docarray/docarray: docs/user_guide/sending/ser/send_doc.md

# BaseDoc

> In order to send or store [BaseDoc][docarray.base_doc.doc.BaseDoc] you need to serialize them first. 

Reviewer: ```suggestion
You need to serialize a [BaseDoc][docarray.base_doc.doc.BaseDoc] before you can store or send it.
```

!! note
    [BaseDoc][docarray.base_doc.doc.BaseDoc] supports serialization to `protobuf` and `json` formats.

## Serialization to protobuf

You can use [`to_protobuf`][docarray.base_doc.doc.BaseDoc.to_protobuf] to serialize a [BaseDoc][docarray.base_doc.doc.BaseDoc] to a protobuf message object
and use [`from_protobuf`][docarray.base_doc.doc.BaseDoc.from_protobuf] to deserialize it.

```python
from typing import List
from docarray import BaseDoc

class MyDoc(BaseDoc):
    text: str
    tags: List[str]

doc = MyDoc(text='hello world', tags=['hello', 'world'])
proto_message = doc.to_protobuf()
new_doc = MyDoc.from_protobuf(proto_message)
assert doc == new_doc  # True
```

> ## Serialization to json

Reviewer: ```suggestion
## Serialization to JSON
```

Reviewer: Please fix throughout PR
